tools/lock_inspect/evm_connection.rb

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env ruby
+
+require 'pg'
+require 'yaml'
+require 'optparse'
+
+def error_exit(e, conn, error_message = '', exit_code = 1)
+  puts e.message
+  puts error_message unless error_message.empty?
+  conn.close if conn
+  exit(exit_code)
+end
+
+def process_app_name(app_name)
+  if app_name.end_with?('...')
+    raise "The application name for MIQ worker: {#app_name} is truncated"
+  end
+  items = app_name.split(/[\[\]\s,]+/)
+  {
+    :pid                  => items[1].to_i,
+    :class_name           => items[2],
+    :compressed_id        => items[3].to_i,
+    :server_compressed_id => items[5].to_i,
+    :zone_name            => items[6],
+    :zone_compressed_id   => items[7].to_i
+  }
+end
+
+default_output = "evm_connection.yml"
+default_server = "127.0.0.1"
+default_port = 5432
+default_user = nil
+default_password = nil
+
+options = {
+  :output   => default_output,
+  :server   => default_server,
+  :port     => default_port,
+  :user     => default_user,
+  :password => default_password
+}
+
+OptionParser.new do |opts|
+  opts.banner = <<-BANNER
+Usage: evm_connection.rb [options] [output_file | #{default_output}]
+Output evm connection information in YAML format.
+  BANNER
+
+  opts.separator ""
+  opts.separator "Specific options:"
+
+  opts.on("-s", "--server [STRING]",
+          "The postgre server address, default to #{default_server}") do |server|
+    options[:server] = server
+  end
+
+  opts.on("-p", "--port [INTEGER]",
+          "The postgre server port, default to #{default_port}") do |port|
+    options[:port] = port
+  end
+
+  opts.on("-u", "--user [STRING]",
+          "The user to login to postgre server, default to current user") do |user|
+    options[:user] = user
+  end
+
+  opts.on("-a", "--password [STRING]",
+          "The password to login") do |password|
+    options[:password] = password
+  end
+
+  opts.on_tail("-h", "--help", "Show this message") do
+    puts opts
+    exit
+  end
+end.parse!
+output = ARGV[0] || options[:output]
+
+puts "Exporting evm connection information to #{output} ..."
+output_file = File.open(output, 'w')
+
+conn_options = {
+  :dbname => 'vmdb_production',
+  :host   => options[:server],
+  :port   => options[:port]
+}
+if options[:user]
+  conn_options[:user] = options[:user]
+end
+if options[:password]
+  conn_options[:password] = options[:password]
+end
+begin
+  conn = PG::Connection.open(conn_options)
+rescue PG::Error => e
+  error_exit(e, conn, "Can't connect to vmdb_production database.")
+end
+
+begin
+  res = conn.exec_params('SELECT application_name FROM pg_stat_activity ORDER BY application_name;')
```

Comment:
The idea is this YAML file can be processed in the later steps so we can try to correlate behavior with the cause of bad behavior.